flocker/node/agents/gce.py

```diff
@@ -684,17 +707,55 @@ def detach_volume(self, blockdevice_id):
             # this point). Might involve putting all GCE interactions behind a
             # zope interface and then using a proxy implementation to inject
             # code.
-            self._do_blocking_operation(
+            result = self._do_blocking_operation(
                 self._compute.instances().detachDisk,
                 instance=attached_to,
                 deviceName=blockdevice_id,
                 timeout_sec=VOLUME_DETATCH_TIMEOUT
             )
+
+            # If there is an outstanding detach operation, attach_to will be
+            # reported, but an attempt to detach will quickly fail with an
+            # `INVALID_FIELD_VALUE`.
+            #
+            # Attempt to detect this scenario, and poll until the volume is
+            # detached.
```

Comment:
:grey_question: Makes sense. We discussed whether the same problem might affect attach_volume for volumes that are attaching and delete_volume for volumes that are deleting. But it sounds like the problem only occurs in cleanup code.